frb_codegen/src/parser/ty.rs

```diff
@@ -192,6 +192,40 @@ impl<'a> TypeParser<'a> {
                         IrTypePrimitive::U8 => Some(SyncReturn(IrTypeSyncReturn::VecU8)),
                         _ => None,
                     },
+                    Some(Optional(option)) => match *option.inner {
+                        Primitive(primitive) => Some(SyncReturn(IrTypeSyncReturn::Option(
+                            Box::new(IrTypeSyncReturn::Primitive(primitive)),
+                        ))),
+                        Delegate(IrTypeDelegate::String) => Some(SyncReturn(
+                            IrTypeSyncReturn::Option(Box::new(IrTypeSyncReturn::String)),
+                        )),
+                        PrimitiveList(primitive) => match primitive.primitive {
+                            IrTypePrimitive::U8 => Some(SyncReturn(IrTypeSyncReturn::Option(
+                                Box::new(IrTypeSyncReturn::VecU8),
+                            ))),
+                            _ => None,
+                        },
+                        Boxed(inner)
+                            if inner.inner.is_rust_opaque() || inner.inner.is_dart_opaque() =>
+                        {
+                            match *inner.inner {
+                                RustOpaque(opaque) => Some(SyncReturn(IrTypeSyncReturn::Option(
+                                    Box::new(IrTypeSyncReturn::RustOpaque(opaque)),
+                                ))),
+                                DartOpaque(opaque) => Some(SyncReturn(IrTypeSyncReturn::Option(
+                                    Box::new(IrTypeSyncReturn::DartOpaque(opaque)),
+                                ))),
+                                _ => None,
+                            }
+                        }
+                        _ => None,
+                    },
```

Comment:
Maybe it is possible to use existing code for non-Option cases instead of duplicating. For example, extract a function